libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccount.m

```diff
@@ -97,198 +96,273 @@ - (instancetype)initWithCredentials:(SFOAuthCredentials*) credentials {
-    }
-    
-    return _accountIdentity;
+    __block SFUserAccountIdentity *identity = nil;
+    dispatch_sync(_syncQueue, ^{
+        identity = self->_accountIdentity;
+    });
+    return identity;
 }
 
 - (NSURL*)apiUrl {
-    if (self.communityId) {
-        NSURL *communityUrl = [self communityUrlWithId:self.communityId];
-        if (communityUrl) {
-            return communityUrl;
-        }
+    NSString *communityId = _communityId;
+    __block NSURL *url = _credentials.apiUrl;
+    if (communityId) {
+        dispatch_sync(_syncQueue, ^{
+            NSURL *communityUrl = [self communityUrlWithId:communityId];
+            if (communityUrl) {
+                url =  communityUrl;
+            }
+        });
     }
-    return self.credentials.apiUrl;
+    return url;
+}
+
+- (NSString *)email {
+    __block NSString *email = nil;
+    SFIdentityData *data = _idData;
+    dispatch_sync(_syncQueue, ^{
+        email = data.email;
+    });
+    return email;
+}
+
+- (NSString *)fullName {
+    __block NSString *fullName = nil;
+    SFIdentityData *data = _idData;
+    dispatch_sync(_syncQueue, ^{
+        fullName = [NSString stringWithFormat:@"%@ %@",data.firstName,data.lastName];
+    });
+    return fullName;
+}
+
+- (NSString *)userName {
+    __block NSString *userName = nil;
+    SFIdentityData *data = _idData;
+    dispatch_sync(_syncQueue, ^{
+        userName = [NSString stringWithFormat:@"%@",data.username];
+    });
+    return userName;
 }
 
 - (NSURL*)communityUrlWithId:(NSString *)communityId {
     if (!communityId) {
         return nil;
     }
-    
-    SFCommunityData *info = [self communityWithId:communityId];
-    return info.siteUrl;
+    __block NSURL *siteUrl = nil;
+    dispatch_sync(_syncQueue, ^{
+        SFCommunityData *info = [self communityWithId:communityId];
+        siteUrl = info.siteUrl;
+    });
+    return siteUrl;
 }
 
 - (SFCommunityData*)communityWithId:(NSString*)communityId {
-    for (SFCommunityData *info in self.communities) {
-        if ([info.entityId isEqualToString:communityId]) {
-            return info;
+    if (!communityId || !_communities) return nil;
+    
+    __block SFCommunityData *communityData = nil;
+    __block  NSArray<SFCommunityData *> *communitiesCopy = [_communities copy];
+    dispatch_sync(_syncQueue, ^{
+        for (SFCommunityData *info in communitiesCopy) {
+            if ([info.entityId isEqualToString:communityId]) {
+                communityData = info;
+                break;
+            }
         }
-    }
-    return nil;
+    });
+    return communityData;
+}
+
+- (NSArray<SFCommunityData *> *)communities {
+    __block NSArray<SFCommunityData *> *communitiesLocal = nil;
+    dispatch_sync(_syncQueue, ^{
+        communitiesLocal = self->_communities;
+    });
+    return communitiesLocal;
+}
+
+- (void)setCommunities:(NSArray<SFCommunityData *> *)communities {
```

Comment:
For `8.0`, this should also be removed (we spoke about it earlier). This property that backs `SFCommunityData` should be moved to an extension in the app code. It has no place in the `Mobile SDK`, since we don't populate it, nor do we use it. It satisfies an app specific use case.

Reply:
@bhariharan We should then deprecate communityUrlWithId, communityWithId too. What do u think?

Reply:
Sounds good.